test/expression.cpp

```diff
@@ -59,35 +67,40 @@ void test_eval() {
     // We test on a binary operator
     {
         std::unordered_map<std::string, T> in{{"x", T(0.125)}};
-        REQUIRE(eval(x/T(0.125), in) == 1.);
+        REQUIRE(eval(x / T(0.125), in) == 1.);
     }
     // We test on a deeper tree
     {
-        expression ex = (x*y + x*y*y*y)/(x*y + x*y*y*y) - x/y;
+        expression ex = (x * y + x * y * y * y) / (x * y + x * y * y * y) - x / y;
         std::unordered_map<std::string, T> in{{"x", T(0.125)}, {"y", T(0.125)}};
         REQUIRE(eval(ex, in) == 0.);
     }
     // We test the corner case of a dictionary not containing the variable.
     {
-        expression ex = x*y;
+        expression ex = x * y;
         std::unordered_map<std::string, T> in{{"x", T(0.125)}};
         REQUIRE_THROWS(eval(ex, in));
     }
     // We test the implementation of all function.
     {
         std::unordered_map<std::string, T> in{{"x", T(0.125)}};
+        REQUIRE(eval(acos(x), in) == acos(T(0.125)));
+        //REQUIRE(eval(acosh(x+heyoka::expression(T(1.))), in) == acosh(T(1.125)));
+        REQUIRE(eval(asin(x), in) == asin(T(0.125)));
+        //REQUIRE(eval(asinh(x), in) == asinh(T(0.125)));
```

Comment:
These are commented out the tests fail (asinh already in double, 1e-17 diff) ..... I cannot see reasons that are not platform dependent ...

Reply:
```/home/dario/Develop/heyoka/test/expression.cpp:92: FAILED:
  REQUIRE( eval(asinh(x), in) == asinh(T(0.125)) )
with expansion:
  0.1246767469 == 0.1246767469
```
difference is:  -1.38778e-17